extensions/src/main/java/org/pentaho/platform/plugin/services/security/userrole/oauth/PentahoClientRegistrationFactory.java

```diff
@@ -0,0 +1,105 @@
+/*! ******************************************************************************
+ *
+ * Pentaho
+ *
+ * Copyright (C) 2024 by Hitachi Vantara, LLC : http://www.pentaho.com
+ *
+ * Use of this software is governed by the Business Source License included
+ * in the LICENSE.TXT file.
+ *
+ * Change Date: 2029-07-20
+ ******************************************************************************/
+
+package org.pentaho.platform.plugin.services.security.userrole.oauth;
+
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.factory.FactoryBean;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.core.AuthenticationMethod;
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
+import org.springframework.security.oauth2.core.ClientAuthenticationMethod;
+
+import java.util.Map;
+import java.util.Objects;
+
+public class PentahoClientRegistrationFactory implements FactoryBean<ClientRegistration> {
+
+  private Map<String, String> propertiesMap;
+  private Map<String, Object> configurationMetadata;
+
+  public PentahoClientRegistrationFactory( Map<String, String> propertiesMap ) {
+    this.propertiesMap = propertiesMap;
+  }
+
+  public PentahoClientRegistrationFactory( Map<String, String> propertiesMap,
+                                           Map<String, Object> configurationMetadata ) {
+    this.propertiesMap = propertiesMap;
+    this.configurationMetadata = configurationMetadata;
+  }
+
+  @Override
+  public ClientRegistration getObject() {
+    ClientRegistration.Builder clientRegistrationBuilder =
+      ClientRegistration.withRegistrationId( propertiesMap.get( "registrationId" ) );
+
+    if ( propertiesMap.containsKey( "clientId" ) ) {
+      clientRegistrationBuilder.clientId( propertiesMap.get( "clientId" ) );
+    }
+    if ( propertiesMap.containsKey( "clientSecret" ) ) {
+      clientRegistrationBuilder.clientSecret( propertiesMap.get( "clientSecret" ) );
```

Comment:
This should be a call to decrypt the property.

Reply:
addressed with pentaho `Encr.java`